recipes/box2d/all/conanfile.py

```diff
@@ -29,27 +24,33 @@ def configure(self):
             del self.options.fPIC
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("box2d-%s" % self.version, self._source_subfolder)
+        tools.files.get(self,
+                        **self.conan_data["sources"][self.version],
+                        destination=self.source_folder,
+                        strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BOX2D_BUILD_SHARED"] = self.options.shared
+        tc.variables["BOX2D_BUILD_STATIC"] = not self.options.shared
+        if self.settings.os == "Windows" and self.options.shared:
+            tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        tc.variables["BOX2D_BUILD_EXAMPLES"] = False
+        tc.generate()
 
     def build(self):
         cmake = CMake(self)
-        cmake.definitions["BOX2D_BUILD_SHARED"] = self.options.shared
-        cmake.definitions["BOX2D_BUILD_STATIC"] = not self.options.shared
-        if self.settings.os == "Windows" and self.options.shared:
-            cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
-        cmake.definitions["BOX2D_BUILD_EXAMPLES"] = False
-        cmake.configure()
+        cmake.configure(build_script_folder="Box2D")
         cmake.build()
 
     def package(self):
-        self.copy("License.txt", dst="licenses", src=os.path.join(self._source_subfolder, "Box2D"))
-        self.copy("*.h", dst=os.path.join("include", "Box2D"), src=os.path.join(self._source_subfolder, "Box2D", "Box2D"))
-        self.copy("*.lib", dst="lib", keep_path=False)
-        self.copy("*.dll", dst="bin", keep_path=False)
-        self.copy("*.so*", dst="lib", keep_path=False, symlinks=True)
-        self.copy("*.dylib", dst="lib", keep_path=False)
-        self.copy("*.a", dst="lib", keep_path=False)
+        tools.files.copy(self, "License.txt", src=os.path.join(self.source_folder, "Box2D"), dst=os.path.join(self.package_folder,"licenses"))
+        tools.files.copy(self, "*.h", src=os.path.join(self.source_folder, "Box2D", "Box2D"), dst=os.path.join(self.package_folder, "include", "Box2D"))
+        tools.files.copy(self, "*.lib", src=os.path.join(self.source_folder, "Box2D"), dst=os.path.join(self.package_folder, "lib"), keep_path=False)
+        tools.files.copy(self, "*.dll", src=os.path.join(self.source_folder, "Box2D"), dst=os.path.join(self.package_folder, "bin"), keep_path=False)
+        tools.files.copy(self, "*.so*", src=os.path.join(self.source_folder, "Box2D"), dst=os.path.join(self.package_folder, "lib"), keep_path=False)
+        tools.files.copy(self, "*.dylib", src=os.path.join(self.source_folder, "Box2D"), dst=os.path.join(self.package_folder, "lib"), keep_path=False)
+        tools.files.copy(self, "*.a", src=os.path.join(self.source_folder, "Box2D"), dst=os.path.join(self.package_folder, "lib"), keep_path=False)
```

Comment:
Confusing. It works because you are doing a in-source build.

Reply:
I still don't have an opinion regarding layout as I have not tried it enough in the new recipes. I agree that this block should be easier to write and maybe the layout and folders help with the path management